test/test/recovery/RecoveryTests.java

```diff
@@ -30,7 +30,15 @@ public void stringBuilder(TestProcess p) throws Exception {
         Assert.isLess(out.ratio("unknown|break_compiled"), 0.002);
     }
 
-    @Test(mainClass = StringBuilderTest.class, debugNonSafepoints = true, arch = {Arch.ARM64, Arch.ARM32})
+    @Test(
+        mainClass = StringBuilderTest.class,
+        debugNonSafepoints = true,
+        arch = {Arch.ARM64, Arch.ARM32},
+        // C2 Inlining can cause some issues when aggressive inlining happens
```

Comment:
If the problem is inlining, why can't we run this with `-XX:-Inline` on the versions affected by this behavior?

Reply:
> Let's not hide bugs here

I'm not proposing to hide it. I'm saying that instead of disabling the test altogether on modern JDK versions running on ARM, we could think of having a separate test which runs with `-XX:-Inline`, if that really solves the problem. It could still help us catching possible regressions, especially since there isn't a clear timeline towards a fix.

Reply:
The comment sounds hand-waving. How can you claim something about aggressive inlining if you haven't got to the root cause yet?
You can frame it like "C2 often loses PcDesc mapping from arraycopy intrinsic to the original bytecode."
Link JBS issue when you create it.

Reply:
Testing with `-XX:-Inline` has no practical benefit.